src/platform/forms/components/review/SavableErrorMessage.jsx

```diff
@@ -0,0 +1,119 @@
+// libs
+import React from 'react';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+
+// platform - forms - actions
+import {
+  saveErrors,
+  saveAndRedirectToReturnUrl as saveAndRedirectToReturnUrlAction,
+} from 'platform/forms/save-in-progress/actions';
+import { toggleLoginModal as toggleLoginModalAction } from 'platform/site-wide/user-nav/actions';
+
+import { APP_TYPE_DEFAULT } from 'platform/forms-system/src/js/constants';
+
+// platform - forms components
+import ErrorMessage from 'platform/forms/components/common/alerts/ErrorMessage';
+
+// platform - forms containers
+import SaveFormLink from 'platform/forms/save-in-progress/SaveFormLink';
+
+// platform
+import CallHRC from 'platform/static-data/CallHRC';
+
+// platform - forms - selectors
+import {
+  formSelector,
+  showLoginModalSelector,
+  userSelector,
+} from 'platform/forms/selectors/review';
+import { defaultProps } from 'recompose';
+
+function SavableErrorMessage(props) {
+  const { route, formConfig, user, form, location, showLoginModal } = props;
+
+  const savedStatus = form?.savedStatus;
+  const appType = formConfig?.customText?.appType || APP_TYPE_DEFAULT;
+  const CustomSubmissionError = formConfig?.submissionError;
+  const errorText = formConfig?.errorText;
+
+  const saveLink = (
+    <SaveFormLink
+      locationPathname={location?.pathname}
+      form={form}
+      formConfig={route?.formConfig}
+      user={user}
+      showLoginModal={showLoginModal}
+      saveAndRedirectToReturnUrl={defaultProps.saveAndRedirectToReturnUrl}
+      toggleLoginModal={defaultProps.toggleLoginModal}
```

Comment:
Is this how `defaultProps` is used? I've never used it myself, but it doesn't look the same as the [API docs](https://github.com/acdlite/recompose/blob/master/docs/API.md#defaultprops). Looks like `defaultProps` is a function used in conjunction with [`compose`](https://github.com/acdlite/recompose/blob/master/docs/API.md#compose) to create higher-order components. :confused:

Reply:
According to the [`recompose` project page](https://github.com/acdlite/recompose), it looks like `recompose` is depreciated in favor of hooks.

>Hooks solves all the problems I attempted to address with Recompose three years ago, and more on top of that. I will be discontinuing active maintenance of this package (excluding perhaps bugfixes or patches for compatibility with future React releases), and recommending that people use Hooks instead.

How would we use hooks here instead? Do we need to? Should these `defaultProps` be `props`?

Is this tested? (I realize this gets into the weird space of component integration testing.)

Reply:
I think this was an auto import. I have removed this as it was not intended.